docs/source/chainerx/install/index.rst

```diff
@@ -9,47 +9,81 @@ ChainerX, or ``chainerx``, can be installed as a top level Python package along
+
+You can specify either of the following environment variables to specify where to look for cuDNN installation.
+
+=========================== ========================================================================================================
+Environment variable        Description
+=========================== ========================================================================================================
+``CUDNN_ROOT_DIR``          Path to your cuDNN installation.
+``CHAINERX_CUDNN_USE_CUPY`` ``1`` to search for cuDNN library and include files in existing `CuPy <https://docs-cupy.chainer.org/>`_
+                            installation.
+                            Only applicable for CuPy installed via wheel (binary) distribution.
+                            Other variables related to cuDNN paths (such as ``CUDNN_ROOT_DIR``) are ignored.
+                            Be warned that the resulting executable will be invalidated if CuPy is uninstalled, moved or
+                            replaced.
+=========================== ========================================================================================================
+
+To support the :ref:`NumPy/CuPy fallback <chainerx-tutorial-numpy-cupy-fallback>` mechanism, currently ChainerX with the CUDA support requires CuPy to be installed together.
+
 
-Example
-~~~~~~~
+.. seealso::
 
-For instance, to install ChainerX without CUDA support, run the following:
+    `CuPy installation guide <https://docs-cupy.chainer.org/en/stable/install.html>`_
+
+
+.. _chainerx-install-examples:
+
+Examples
+--------
+
+Install ChainerX without CUDA support:
 
 .. code-block:: console
 
     $ export CHAINER_BUILD_CHAINERX=1
     $ export MAKEFLAGS=-j8  # Using 8 parallel jobs.
     $ pip install --pre chainer
 
-CUDA support
-------------
 
-When installing with the CUDA support, you also need to specify the cuDNN installation path since CUDA without cuDNN is currently not supported.
+Install ChainerX depending on CuPy wheel distribution:
 
-To support the :ref:`NumPy/CuPy fallback <chainerx-tutorial-numpy-cupy-fallback>` mechanism, currently ChainerX with the CUDA support requires CuPy to be installed together.
+.. code-block:: console
+
+    $ pip install --pre cupy_cuda101  # Note: Choose the proper CUDA SDK version number.
+    $ export CHAINER_BUILD_CHAINERX=1
+    $ export CHAINERX_BUILD_CUDA=1
+    $ export CHAINERX_CUDNN_USE_CUPY=1
+    $ export MAKEFLAGS=-j8  # Using 8 parallel jobs.
+    $ pip install --pre chainer
 
-.. note::
 
-    For ChainerX, we suggest that you do not install CuPy with a CuPy wheel (precompiled binary) package because it contains a cuDNN library.
-    Installation would fail if the versions of the cuDNN library contained in the CuPy wheel package and the one specified in `CUDNN_ROOT_DIR` were different.
+Install ChainerX with CuPy built from source:
 
 .. code-block:: console
```

Comment:
Do we still need this install instruction below?
I think `CHAINERX_CUDNN_USE_CUPY` should be the next standard way to build with ChainerX.

Reply:
I assume non-negligible proportion of developers build CuPy from source.
In fact, it's adviced to align CuPy version to that of Chainer: those who build ChainerX from master branch should also use CuPy master branch. I think this example is informative for those developers.

Reply:
Seems reasonable